chef/cookbooks/provisioner/templates/suse/crowbar_register.erb

```diff
@@ -46,9 +46,16 @@ while test $# -gt 0; do
         -h|--help|--usage|-\?) usage ;;
         -v|--verbose) CROWBAR_VERBOSE=1 ;;
         -f|--force) CROWBAR_FORCE=1 ;;
-        -k|--keep-existing-hostname) KEEP_EXISTING_HOSTNAME=1 ;;
         --gpg-auto-import-keys) CROWBAR_AUTO_IMPORT_KEYS=1 ;;
         --no-gpg-checks) CROWBAR_NO_GPG_CHECKS=1 ;;
+        -k|--keep-existing-hostname)
+            if test $# -eq 1; then
```

Comment:
Unfortunately, this bit (intended to make the 0/1 optional) only works if there's no option after this one. Maybe check if the next option is 0/1/true/false and consume it as a value for KEEP_EXISTING_HOSTNAME only if that's the case?